integration-tests/parser/indentation/pipe-needing-indentation/test.test.js.md

```diff
@@ -0,0 +1,27 @@
+# Snapshot report for `integration-tests/parser/indentation/pipe-needing-indentation/test.test.js`
+
+The actual snapshot is saved in `test.test.js.snap`.
+
+Generated by [AVA](https://avajs.dev).
+
+## Invocation
+
+    'elm-in-elm -m src/Main.elm'
+
+## Stderr
+
+    `Compiled in DEV mode. Follow the advice at https://elm-lang.org/0.19.1/optimize for better performance and smaller assets.␊
+    ␊
+    ---------------------------␊
+    -- COMPILER ERROR ---------␊
+    ---------------------------␊
+    Couldn't find the value `main` in the main module given to the compiler!␊
```

Comment:
Not ideal, huh? :smile:

Reply:
True! I didn't actual take care to read this message I was too concerned about the exit code.

Reply:
With `P.End` in `module_` I get:

```
---------------------------
-- STARTING THE COMPILER --
---------------------------

---------------------------
-- COMPILER ERROR ---------
---------------------------
Parse problem: ExpectingEnd
  --> src/Main.elm:6:1
   | 
6  | | Nothing
   | ^ ExpectingEnd
```

Reply:
The issue is we have

```elm
                |= P.oneOf
                    [ P.succeed (\newCtor -> P.Loop (newCtor :: reversedCtors))
                        |. onlyIndented (P.token (P.Token "|" ExpectingPipe))
                        |. P.spaces
                        |= onlyIndented constructor
                    , P.succeed (P.Done (List.reverse reversedCtors))
                    ]
    ```

and we do not commit to the "|".

Reply:
and so we need to rework the `onlyIdented` function to commit if the parser would succeed (regardless of indentation).

```elm
{-| Fail if current column <= 1 (these are 1-based, so 1 is leftmost.)
-}
onlyIndented : Parser_ a -> Parser_ a
onlyIndented parser =
    P.succeed identity
        |. checkIndent (\_ column -> column > 1) ExpectingIndentation
        |= parser
```

@Janiczek do you know how we would do that. Relevant: https://github.com/elm/parser/blob/master/semantics.md

Reply:
We talked about this here: https://discord.com/channels/578305644780716039/578306156972212224/728362159288877117

> need to rework the onlyIdented function to commit if the parser would succeed (regardless of indentation).

This is false! The problem **isn't** that the custom type parser did not commit to the "|". Instead the problem was that the `declarations` parser was happy to stop parsing and return success as soon as it encountered an error!